### PR TITLE
Open Api Spec Filter

### PR DESCRIPTION
**Pull request by gracekarina**
This PR, completes the implementation of Filters in the inflector.

### CODE REVIEW

File: src/main/java/io/swagger/oas/inflector/utils/DefaultSpecFilter.java [hunk 1]
```diff
@@ -6,39 +6,40 @@
...
 
 public class DefaultSpecFilter extends AbstractSpecFilter {
 
-    /*@Override
-    public boolean isOperationAllowed(Operation operation, ApiDescription api, Map<String, List<String>> params, Map<String, String> cookies, Map<String, List<String>> headers) {
+    @Override
+    public Optional<Operation> filterOperation(Operation operation, ApiDescription api, Map<String, List<String>> params, Map<String, String> cookies, Map<String, List<String>> headers) {
         if(operation.getExtensions() != null && operation.getExtensions().containsKey(Constants.X_INFLECTOR_HIDDEN)) {
-            return false;
+            return Optional.of(operation);
```
Comment (frantuma) [inline src/main/java/io/swagger/oas/inflector/utils/DefaultSpecFilter.java]: I guess you want to return `Optional.empty()` here

File: src/main/java/io/swagger/oas/inflector/utils/DefaultSpecFilter.java [hunk 2]
```diff
@@ -6,39 +6,40 @@
...
-        return true;
+        return Optional.of(operation);
     }
 
     @Override
-    public boolean isParamAllowed(Parameter parameter, Operation operation, ApiDescription api, Map<String, List<String>> params, Map<String, String> cookies, Map<String, List<String>> headers) {
+    public Optional<Parameter> filterParameter(Parameter parameter, Operation operation, ApiDescription api, Map<String, List<String>> params, Map<String, String> cookies, Map<String, List<String>> headers) {
         if(parameter.getExtensions() != null && parameter.getExtensions().containsKey(Constants.X_INFLECTOR_HIDDEN)) {
-            return false;
+            return Optional.of(parameter);
```
Comment (frantuma) [inline src/main/java/io/swagger/oas/inflector/utils/DefaultSpecFilter.java]: same as above

File: src/main/java/io/swagger/oas/inflector/utils/DefaultSpecFilter.java [hunk 3]
```diff
@@ -6,39 +6,40 @@
...
-        return true;
+        return Optional.of(parameter);
     }
 
     @Override
-    public boolean isPropertyAllowed(Schema model, Schema property, String propertyName, Map<String, List<String>> params, Map<String, String> cookies, Map<String, List<String>> headers) {
+    public Optional<Schema> filterSchemaProperty(Schema model, Schema property, String propertyName, Map<String, List<String>> params, Map<String, String> cookies, Map<String, List<String>> headers) {
         if(property.getExtensions() != null && property.getExtensions().containsKey(Constants.X_INFLECTOR_HIDDEN)) {
-            return false;
+            return  Optional.of(property);
```
Comment (frantuma) [inline src/main/java/io/swagger/oas/inflector/utils/DefaultSpecFilter.java]: same as above

File: src/main/java/io/swagger/oas/inflector/utils/DefaultSpecFilter.java (hunk 1 above)
Comment (gracekarina) [reply src/main/java/io/swagger/oas/inflector/utils/DefaultSpecFilter.java]: Yes. I've change it. Thanks!

File: src/main/java/io/swagger/oas/inflector/utils/DefaultSpecFilter.java (hunk 2 above)
Comment (gracekarina) [reply src/main/java/io/swagger/oas/inflector/utils/DefaultSpecFilter.java]: changed it!

File: src/main/java/io/swagger/oas/inflector/utils/DefaultSpecFilter.java (hunk 3 above)
Comment (gracekarina) [reply src/main/java/io/swagger/oas/inflector/utils/DefaultSpecFilter.java]: Changed it

**Pull request merged**